src/components/pages/group/common/icon-checkbox-field/checkbox.tsx

```diff
@@ -7,43 +7,53 @@ export interface ITraitInputProps {
   type?: string;
   values: any[];
   gridColumns?;
+  isReadOnly?: boolean;
   onUpdate;
   defaultValue?;
 }
 
 const VH: any = VisuallyHidden;
 
-const CustomCheckbox = ({ value, label, icon, type, ...props }) => (
+const CustomCheckbox = ({ value, label, icon, type, isReadOnly, ...props }) => (
   <label
     role="checkbox"
     className="custom-checkbox"
     style={{ padding: "0.25rem" }}
     aria-checked={props.isChecked}
   >
-    <VH as="input" type="checkbox" {...props} value={value} />
+    <VH disabled={isReadOnly} as="input" type="checkbox" {...props} value={value} />
     <Tooltip title={icon} placement="top" hasArrow={true}>
       <Image size="3rem" ignoreFallback={true} src={getLocalIcon(icon, type)} alt={icon} />
     </Tooltip>
   </label>
 );
 
-const CheckBoxItems = ({ values, type, onUpdate, defaultValue }: ITraitInputProps) => (
-  <CheckboxGroup
-    defaultValue={defaultValue && defaultValue.map((o) => o.toString())}
-    onChange={(v) => onUpdate(v.map((i) => Number(i)))}
-    isInline={true}
-    spacing={3}
-  >
-    {values.map((o) => (
-      <CustomCheckbox
-        key={o.id}
-        value={o.id.toString()}
-        label={o.value}
-        icon={o.name}
-        type={type}
-      />
-    ))}
-  </CheckboxGroup>
-);
+const CheckBoxItems = ({ values, type, onUpdate, isReadOnly, defaultValue }: ITraitInputProps) => {
+  const canRender = (id) => (isReadOnly ? defaultValue.includes(id) : true);
```

Comment:
Something feels wrong here, avoid function calling and send filtered values from options instead